src/main/java/au/com/agiledigital/idea_search/dao/FedexIdeaDao.java

```diff
@@ -0,0 +1,281 @@
+        this.ao = ao;
+        this.userAccessor = userAccessor;
+    }
+
+    /**
+     * Create new entry to represent the a FedexIdea
+     *
+     * @param fedexIdea FedexIdea model object
+     * @return FedexIdea object created in data store
+     */
+    public FedexIdea create(FedexIdea fedexIdea) {
+        AoFedexIdea aoFedexIdea = this.ao.create(AO_FEDEX_IDEA_TYPE);
+
+        this.prepareAOFedexIdea(aoFedexIdea, fedexIdea);
+
+        // Save the changes to the active object
+        aoFedexIdea.save();
+        // Set the relation of the technology to the idea and save
+        SetTechnology(getAoFedexTechnologies(fedexIdea), aoFedexIdea);
+
+        return this.asFedexIdea(aoFedexIdea);
+    }
+
+    /**
+     * Update a FedexIdea by the page content id
+     *
+     * @param fedexIdea New FedexIdea for data store
+     * @param contentId of the page containing the idea
+     * @return FedexIdea saved to the data store
+     */
+    public FedexIdea updateByContentId(FedexIdea fedexIdea, long contentId) {
+        AoFedexIdea[] aoFedexIdeas =
+                this.ao.find(
+                        AO_FEDEX_IDEA_TYPE,
+                        Query.select().where("CONTENT_ID = ?", contentId)
+                );
+        log.error(String.valueOf(aoFedexIdeas.length));
+        // If the title of the page already exists the create event fails and a page update event is fired on a successful save.
+        // If this happens the aoFedexIdea will not exist in the data store and will need to be created.
+        AoFedexIdea aoFedexIdea = aoFedexIdeas.length == 0 ? this.ao.create(AO_FEDEX_IDEA_TYPE) : aoFedexIdeas[0];
+
+
+        List<AoFedexTechnology> aoFedexTechnology = Arrays.asList(
+                aoFedexIdea.getTechnology()
+        );
+
+        // Deletes the technology item from the table
+        aoFedexTechnology.forEach(this.ao::delete);
+
+        List<AoFedexTechnology> aoTechList = getAoFedexTechnologies(fedexIdea);
+
+        this.prepareAOFedexIdea(aoFedexIdea, fedexIdea);
+        SetTechnology(aoTechList, aoFedexIdea);
+        aoFedexIdea.save();
+
+        return this.asFedexIdea(aoFedexIdea);
+    }
+
+    /**
+     * Extract technologies from FedexIdea and create an active object for each
+     *
+     * @param fedexIdea from model
+     * @return List<AoFedexTechnology> unsaved.
+     */
+    @NotNull
+    private List<AoFedexTechnology> getAoFedexTechnologies(FedexIdea fedexIdea) {
+        return fedexIdea
+                .getTechnologies()
+                .stream()
+                .map(tech -> this.prepareAOFedexTechnology(tech.getTechnology()))
+                .collect(Collectors.toList());
+    }
+
+    //    See comment on https://community.atlassian.com/t5/Jira-questions/ActiveObjects-jira/qaq-p/354375
+    //    This means that the setter is done on the Recipient entity(setting the Filter) and because of the way relational databases...
+    private void SetTechnology(
```

Comment:
Lowercase S or is this forced by Confluence?

```suggestion
    private void setTechnology(
```

Reply:
setTechnologies